pkg/resolver/search.go

```diff
@@ -107,6 +131,80 @@ func (s *SearchResult) Uids() {
 	s.resolveUids()
 }
 
+// Build where clause with rbac by combining clusterscoped, namespace scoped and managed cluster access
+func buildRbacWhereClause(ctx context.Context, userrbac *rbac.UserResourceAccess) exp.ExpressionList {
+	return goqu.Or(
+		matchManagedCluster(userrbac.ManagedClusters), // goqu.I("cluster").In([]string{"clusterNames", ....})
+		goqu.And(
+			matchHubCluster(), // goqu.L(`data->>?`, "_hubClusterResource").Eq("true")
+			goqu.Or(
+				matchClusterScopedResources(userrbac.CsResources), // (namespace=null AND apigroup AND kind)
+				matchNamespacedResources(userrbac.NsResources),    // (namespace AND apiproup AND kind)
+			),
+		),
+	)
+}
+
+// inner function to loop through resources and build the where clause
+func loopThroughResources(resources []rbac.Resource) exp.ExpressionList {
+	var whereCsDs exp.ExpressionList // Stores the where clause for cluster scoped resources
+
+	for i, clusterRes := range resources {
+		whereOrDs := []exp.Expression{goqu.COALESCE(goqu.L(`data->>?`, "apigroup"), "").Eq(clusterRes.Apigroup),
+			goqu.L(`data->>?`, "kind_plural").Eq(clusterRes.Kind)}
+
+		// Using this workaround to build the AND-OR combination query in goqu.
+		// Otherwise, by default goqu will AND everything
+		// (apigroup='' AND kind='') OR (apigroup='' AND kind='')
+		if i == 0 {
+			whereCsDs = goqu.And(whereOrDs...) // First time, AND all conditions
+		} else {
+			//Next time onwards, perform OR with the existing conditions
+			whereCsDs = goqu.Or(whereCsDs, goqu.And(whereOrDs...))
+		}
+	}
+	return whereCsDs
+}
+func matchClusterScopedResources(csRes []rbac.Resource) exp.ExpressionList {
+	var whereNsDs exp.ExpressionList
+
+	if len(csRes) > 0 {
+		return goqu.And(goqu.COALESCE(goqu.L(`data->>?`, "namespace"), "").Eq(""),
+			loopThroughResources(csRes))
+
+	}
+	return whereNsDs
+}
```

Comment:
```suggestion
func matchClusterScopedResources(csRes []rbac.Resource) exp.ExpressionList {
	if len(csRes) > 0 {
		return goqu.And(goqu.COALESCE(goqu.L(`data->>?`, "namespace"), "").Eq(""),
			loopThroughResources(csRes))
	}
	return exp.ExpressionList{}
}
```

Reply:
Tried this, but it threw some error. But, found another way.

Reply:
Tried that before - it is not working. But, made a similar change.

Reply:
thanks!